utils/cachingStrategy.ts

```diff
@@ -34,9 +34,8 @@ export const processMetadata = <T>(
 ): void => {
   const metadata = metadataList.map((meta) => meta || '')
 
-  metadata.forEach((m, i) => {
-    const meta = cacheOrFetchMetadata(metadataCache[m], m)
-
+  metadata.forEach(async (m, i) => {
+    const meta = await cacheOrFetchMetadata(metadataCache[m], m)
```

Comment:
I've already had a problem here (series table if I record), and I managed to do without.
are you sure there won't be side issue with this change?

Reply:
I'm sure there won't be a side issue. I have also updated this line to unify the call of the `processSingleMetadata` function.